internal/buildengine/languageplugin/java_plugin.go

```diff
@@ -29,19 +30,71 @@ import (
 	"github.com/TBD54566975/ftl/jvm-runtime/kotlin"
 )
 
+const JavaBuildToolMaven string = "maven"
+const JavaBuildToolGradle string = "gradle"
+
+type JavaConfig struct {
+	BuildTool string `mapstructure:"build-tool"`
+}
+
+func loadJavaConfig(languageConfig any, language string) (JavaConfig, error) {
+	var javaConfig JavaConfig
+	err := mapstructure.Decode(languageConfig, &javaConfig)
+	if err != nil {
+		return JavaConfig{}, fmt.Errorf("failed to decode %s config: %w", language, err)
+	}
+	return javaConfig, nil
+}
+
+// ModuleJavaConfig is language-specific configuration for Java modules.
 type javaPlugin struct {
 	*internalPlugin
 }
 
 var _ = LanguagePlugin(&javaPlugin{})
 
-func newJavaPlugin(ctx context.Context, config moduleconfig.ModuleConfig) *javaPlugin {
-	internal := newInternalPlugin(ctx, config, buildJava)
+func newJavaPlugin(ctx context.Context, language string) *javaPlugin {
+	internal := newInternalPlugin(ctx, language, buildJava)
 	return &javaPlugin{
 		internalPlugin: internal,
 	}
 }
 
+func (p *javaPlugin) ModuleConfigDefaults(ctx context.Context, dir string) (moduleconfig.CustomDefaults, error) {
+	defaults := moduleconfig.CustomDefaults{
+		GeneratedSchemaDir: "src/main/ftl-module-schema",
+		Deploy:             []string{"launch", "quarkus-app"},
+		// Watch defaults to files related to maven and gradle
+		Watch: []string{"pom.xml", "src/**", "build/generated", "target/generated-sources"},
```

Comment:
Previously `watch` was different based on maven/gradle, but I think it's fine to just use the same watch patterns for both